zerver/views/message_edit.py

```diff
@@ -60,6 +60,14 @@ def fill_edit_history_entries(message_history: List[Dict[str, Any]], message: Me
             del entry['prev_rendered_content_version']
             prev_content = entry['prev_content']
             prev_rendered_content = entry['prev_rendered_content']
+            # mypy infers that prev_rendered_content is of type Optional[str]
+            # based on the assignment near the top of this function, because
+            # message.rendered_content is of that type, as of the writing of
+            # this comment. However, after the assignment immediately above,
+            # prev_rendered_content will always be a string (barring bad data
+            # in the database). The assertion below is necessary because mypy
+            # will otherwise complain that an Optional[str] variable is being
+            # used where a str variable is required.
             assert prev_rendered_content is not None
```

Comment:
I don't think the comment is necessary; this is essentially the only way we use the `assert foo is not None` statement in the codebase.

Reply:
I'll remove it then. The slight confusion I had was not about it being a hint to mypy, but why mypy was seeing it as Optional[str]; I initially missed that it was inferring it from that first assignment.